src/types/LeaveNoDust.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+import {IERC20} from "../BaseConditionalOrder.sol";
+
+import "../BaseConditionalOrder.sol";
+
+/**
+ * @title A smart contract that executes a trade at the specified exchange rate for the entirety of the current balance.
+ * @dev Designed to be used with the CoW Protocol Conditional Order Framework.
+ *      This order may be used e.g. if a rebasing token is intended to be fully sold, but its balance keeps changing from 
+ *      block to block making it hard to specify a fixed amount up-front.
+ *      It is recommended to use sell and buy amounts from a verified quote to ensure the trade is likely to execute.
+ *      In case the sell balance at the time of execution differs significantly from the one at the time of order placement,
+ *      additional care in accounting for price impact and the slippage tolerance may be needed.
+ *      In particular, since the price is scaled linearly, however quotes include a non-linear network cost as part of the exchange rate,
+ *      having a significantly lower sell amount, scaled linearly may not allow for sufficient network fee be taken from surplus.
+ */
+contract LeaveNoDust is BaseConditionalOrder {
```

Comment:
BaseConditionalOrder will check during the validation the order matches the expected order.

This will never be the case, if the `getTradeableOrder` yields the order in block `N`, but its verified at `N+1`.

I guess you need to override the validation to allow a margin of error to let it go through. Assuming these tokens won't appreciate more than 20% yearly, you could allow a 0.06% error, or if you want, you can add this error threshold to your Data so its specified by the order creator.